### PR TITLE
8 make threshold and return type configureable

### PR DESCRIPTION
**Pull request by Oliver-Pikett**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,4 @@
 **/target
-files/alexnet_Opset18.onnx
 files/libonnxruntime.dylib
-files/libonnxruntime.1.17.0.dylib
+files/libonnxruntime.1.16.3.dylib
```
Comment (krystal1549) [inline .gitignore]: Make sure you pull in main so that the change we made on your other branch is reflected here

File: src/vad_environment.rs [hunk 2]
```diff
@@ -6,12 +6,13 @@ use std::path::{Path, PathBuf};
 //anyway to get these things to not come up as unsued without doing allow deadcode at start of file?
 pub struct VadEnvironment {
     model_location: PathBuf,
-    //keep vad_sessions in vector?
+    return_probs: bool,
```
Comment (krystal1549) [inline src/vad_environment.rs:9]: Maybe me being silly but could return_probs be named something more informative. I'm not sure what you mean by it

File: src/vad_environment.rs [hunk 3]
```diff
@@ -84,7 +70,7 @@ mod tests {
             }
         }
 
-        let environment_ok = VadEnvironment::new("files/silero_vad.onnx");
+        let environment_ok = VadEnvironment::new("files/silero_vad.onnx", false, 0.5);
```
Comment (krystal1549) [inline src/vad_environment.rs:73]: Pulling in main should also pull in the CI/CD pipeline you added so we should be able to see that these tests work as part of the review which would be nice

File: src/vad_session.rs [hunk 4]
```diff
@@ -121,34 +148,33 @@ impl VadSession {
...
-    if (audio_data.len() % 16000) != 0 {
-        let chunk = &audio_data[start_index..(audio_data.len() as usize)];
-        audio_chunks.push(chunk.to_vec());
+    if (audio_data.len() % sample_rate as usize) != 0 {
+        let window = &audio_data[start_index..(audio_data.len() as usize)];
+        let audio_window: AudioWindow = AudioWindow::new(window.to_vec(), start_index as i32, (audio_data.len()-1).try_into().unwrap());
+        audio_windows.push(audio_window);
     }
-    return audio_chunks;
+    return audio_windows;
```
Comment (krystal1549) [inline src/vad_session.rs:175]: Why has this changed everywhere from audio chunks to windows?

Comment (Oliver-Pikett) [reply src/vad_session.rs:175]: i just thought it was clearer

**Review by SeanEClarke — APPROVED** [review]
Looks good, however some of the naming could be a bit better.

Always try to look at it either as someone picking it up for the first time, of yourself picking it up whilst debugging 6 months down the line. Explicit naming where it is clear the moment it has been read will pay dividends.

 think some of this may be superseded when you do the stateful (and more production like) API.

**Pull request merged**